src/coreclr/jit/morph.cpp

```diff
@@ -6006,22 +6006,25 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
     }
 
 #ifdef FEATURE_SIMD
-    // if this field belongs to simd struct, translate it to simd intrinsic.
-    if (mac == nullptr)
+    if (IsBaselineSimdIsaSupported())
     {
-        GenTree* newTree = fgMorphFieldToSimdGetElement(tree);
-        if (newTree != tree)
+        // if this field belongs to simd struct, translate it to simd intrinsic.
+        if (mac == nullptr)
         {
-            newTree = fgMorphSmpOp(newTree);
-            return newTree;
+            GenTree* newTree = fgMorphFieldToSimdGetElement(tree);
+            if (newTree != tree)
+            {
+                newTree = fgMorphSmpOp(newTree);
+                return newTree;
+            }
         }
-    }
-    else if ((objRef != nullptr) && (objRef->OperGet() == GT_ADDR) && varTypeIsSIMD(objRef->gtGetOp1()))
-    {
-        GenTreeLclVarCommon* lcl = objRef->IsLocalAddrExpr();
-        if (lcl != nullptr)
+        else if ((objRef != nullptr) && (objRef->OperGet() == GT_ADDR) && varTypeIsSIMD(objRef->gtGetOp1()))
         {
-            lvaSetVarDoNotEnregister(lcl->GetLclNum() DEBUGARG(DNER_LocalField));
+            GenTreeLclVarCommon* lcl = objRef->IsLocalAddrExpr();
+            if (lcl != nullptr)
+            {
+                lvaSetVarDoNotEnregister(lcl->GetLclNum() DEBUGARG(DNER_LocalField));
```

Comment:
Can you explain why `else if`-case also needs to be conditional on `IsBaselineSimdIsaSupported`?

Reply:
It probably doesn't "need" to be. Most of the logic here is around ensuring that the SIMD/HWIntrinsic codegen is correct/optimal and isn't strictly needed when we aren't creating SIMD/HWIntrinsic nodes.

I'll update to only cover `fgMorphFieldToSimdGetElement` for simplicity, however.